rre-dataset-generator/dataset_generator.py

```diff
@@ -1,27 +1,86 @@
-import logging
-
-from src.config import Config
+# configuration params
 from src.utils import parse_args
+from src.config import Config
+from src.llm.llm_service import LLMService
+from src.llm.llm_config import LLMConfig
 
-from src.search_engine.solr_search_engine import SolrSearchEngine
+# data structures
+from src.search_engine.data_store import DataStore
+
+# build factories
+from src.search_engine import build_search_engine
+from src.llm import build_chat_model
+from src.writers import build_writer
+
+# logging
 from src.logger import configure_logging
 import logging
 
+
 if __name__ == "__main__":
     args = parse_args()
 
     config = Config.load(args.config_file)
+
     if args.verbose:
         configure_logging(logging.DEBUG)
     else:
         configure_logging(logging.INFO)
+    log = logging.getLogger(__name__)
+
+    search_engine = build_search_engine(search_engine_type=config.search_engine_type,
+                                        endpoint=config.search_engine_collection_endpoint)
+    data_store = DataStore()
+
+    user_queries = []
+    if config.queries is not None:
+        with open(config.queries, 'r', encoding='utf-8') as file:
+            for line in file:
+                if line.strip():
+                    user_queries.append(line.strip())
```

Comment:
We are adding all queries provided by the user into the variable `user_queries`, and we never read from it.
I suggest directly adding all user queries to the datastore (we don't specify the docId, so it probably requires updating the datastore), so we've already have them ready when we calculate the cartesian product

Reply:
changed in the new version by @nicolo-rinaldi